pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/UserProfileCachedDao.java

```diff
@@ -0,0 +1,66 @@
+package org.broadinstitute.ddp.db.dao;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import java.util.Optional;
+import javax.cache.Cache;
+import javax.cache.expiry.Duration;
+
+import org.broadinstitute.ddp.cache.CacheService;
+import org.broadinstitute.ddp.cache.ModelChangeType;
+import org.broadinstitute.ddp.model.user.UserProfile;
+import org.jdbi.v3.core.Handle;
+
+public class UserProfileCachedDao extends SQLObjectWrapper<UserProfileDao> implements UserProfileDao {
+    private static final String CACHE_NAME = "UserProfile.byGuid";
+
+    private static Cache<String, UserProfile> userProfileCache;
+
+    private void initCache() {
+        if (userProfileCache == null || userProfileCache.isClosed()) {
+            userProfileCache = CacheService.getInstance().getOrCreateCache(CACHE_NAME,
+                    new Duration(MINUTES, 10), UserProfileCachedDao.class);
+        }
+    }
+
+    public UserProfileCachedDao(Handle handle) {
+        super(handle, UserProfileDao.class);
+        //  initCache();
+    }
```

Comment:
Sorry. I backed off caching this for now, but probably will add back in.
Just wanted to cache the Authorizations in the TokenConverterFilter. Did have to implement the model modification notifcations.